### ISSUE TITLE
Problem Set Cards: Title (Mobile > Dashboard) 

### ISSUE THREAD

**Issue by alexrcabral** [benetech/MathShare#1405]
Problem Set Cards: Title (Mobile > Dashboard) 

This task defines the sourcing and surfacing of the problem set title. 

## Full Mockup
<img width="374" alt="Mockup of new dashboard, showing the Your Sets page, with rounded pastel problem cards in a single column. Along the top of the page is a Mathshare logo and a profile image. Underneath is the heading, Your Sets, next to a search icon and a dropdown showing the words, Most Recent. Each problem card shows a profile image, the designation Algebra 101, a three dot drop down, the problem set title, and the word Progress, followed by 02 of 06 Problems, and a progress bar beneath." src="https://user-images.githubusercontent.com/47576069/94318289-075f5e80-ff3d-11ea-9822-d3391ddc3432.png">

The above mockup shows the full dashboard in context (Mobile version). 

### Notable changes from mockups: 
- We are not using the "Algebra 101" or similar class/topic designation in this initial mobile release  

## Title 

The title for the set as defined or edited through the problem set creation process should be surfaced on the main portion of the problem set card (towards the middling-top, exactly as shown in the mockups), using the appropriate font size and styling (Source Sans Pro) indicated using the inspection tool on the mockups. 

## Reference
Style Guides: https://invis.io/CQYLLW8J49U
Mobile Mockups: https://invis.io/5HYLLW1G4BD
Fonts: https://www.dropbox.com/sh/c12m6ufweyds20d/AAAdXNPOt8e52DkSAr9pd-a-a/Fonts?dl=0

## Acceptance criteria: 
Given that I'm a logged-in user with sets available, when I access my dashboard, I should be able to view the titles of all my existing problem sets.

**Comment by johnhbenetech** [comment]
@alexrcabral can you say more about  "I should be able to view the titles of all my existing problem sets"

Is this problem sets I've created?, Problem sets I've started to solve?, Both? Are there any differences in the display of these two types of sets? Are there any differences between students and teacher accounts?

Is there any limit to number of sets displayed? If a user has 50 problem sets would we show all 50?

**Comment by alexrcabral** [comment]
@johnhbenetech These are covered in #1401 and #1404 

> All users (students & teachers) who are logged in, and have either created (/edit) or saved (/solve) a set in the past should be shown their existing sets (both /edit and /solve versions), presented in a single column of problem set cards. These are scrollable under the heading "Your Sets", followed by the all existing sets.

> The cards should be rounded rectangles in a single column from the top. Users should be able to scroll through these until they reach the last set (progressive loading five set cards at a time). 

The primary difference is between problem set cards with and without progress, which is the main defining feature between student sets and teacher sets. (See #1407 )

**Comment by johnhbenetech** [comment]
@alexrcabral ok I see. So theoretically they can load infinite problem set (5 at a time). Sounds like a plan

**Comment by alexrcabral** [comment]
There seems to be an issue with the problem set cards, also mentioned in #1401. 

As a logged in user with 10+ existing sets on the current version, none of these are surfaced -- only the button to "Add a Problem Set". 

<img width="401" alt="Screen Shot showing the Mathshare Dashboard with a large button to add your first problem here." src="https://user-images.githubusercontent.com/47576069/95800195-ad37fa80-0cab-11eb-9284-90fbeb599311.png">

If I click this, it shows me three set cards, all titled Solve for X and begins to surface the Example Sets. 

<img width="430" alt="Screen shot showing three pastel problem set cards, all titled solve for x with 4 of 6 problems complete." src="https://user-images.githubusercontent.com/47576069/95800233-ca6cc900-0cab-11eb-9074-1f30254bb357.png">

**Comment by johnhbenetech** [comment]
Next step here is hooking up backend to load sets, then make sure progress is reflected accurately

**Comment by rupeshparab** [comment]
Hooked up backend

**Comment by abaranowska1** [comment]
When I'm a logged-in user with sets available and I access my dashboard, I'm able to view the titles of all my existing problem sets. 
@rupeshparab Can you tell me why this issue is shown twice on the dashboard ? One with progress and one without https://mathshare-qa.diagramcenter.org/#/app/problemSet/view/IJWHAVR7LUPC6
![image](https://user-images.githubusercontent.com/61964176/96846278-f9283380-1451-11eb-9160-cfb79188164b.png)

**Comment by rupeshparab** [comment]
@abaranowska1 
I think you would have created a problem set and also its solution, in this case, you have created a set as a teacher and also solved it as a student.

**Comment by abaranowska1** [comment]
@rupeshparab Yes, exactly. I have created a set as a teacher and also solved it as a student. So I guess teacher should see all the problems?

**Comment by johnhbenetech** [comment]
@abaranowska1 see this comment https://github.com/benetech/MathShare/issues/1405#issuecomment-701532629

**Comment by alexrcabral** [comment]
> should be shown their existing sets (both /edit and /solve versions)

The original intent was to show both edit sets and unowned solve sets, as opposed to in the past how we only surfaced edit sets and their associated work to teachers, while we only surfaced solve versions to students. 

I think this interaction is fine for now, though I'll break out a separate task (#1500  in order to deduplicate solve sets for which the user also owns the edit set.
